crates/store/src/db/sql/mod.rs

```diff
@@ -190,6 +193,317 @@ pub fn select_accounts_by_ids(
+                    b.block_num <= ?2 AND
+                    a.block_num < b.block_num
+            )
+        ORDER BY
+            account_id
+        ",
+    )?;
+
+    let mut select_latest_storage_slot_updates_stmt = conn.prepare_cached(
+        "
+        SELECT
+            account_id, slot, value
+        FROM
+            account_storage_slot_updates a
+        WHERE
+            account_id IN rarray(?1) AND
+            block_num <= ?2 AND
+            NOT EXISTS(
+                SELECT 1
+                FROM account_storage_slot_updates b
+                WHERE
+                    a.account_id = b.account_id AND
+                    b.block_num <= ?2 AND
+                    a.block_num < b.block_num AND
+                    a.slot = b.slot
+            )
+        ORDER BY
+            account_id
+        ",
+    )?;
+
+    let mut select_latest_storage_map_updates_stmt = conn.prepare_cached(
+        "
+        SELECT
+            account_id, slot, key, value
+        FROM
+            account_storage_map_updates a
+        WHERE
+            account_id IN rarray(?1) AND
+            block_num <= ?2 AND
+            NOT EXISTS(
+                SELECT 1
+                FROM account_storage_map_updates b
+                WHERE
+                    a.account_id = b.account_id AND
+                    b.block_num <= ?2 AND
+                    a.block_num < b.block_num AND
+                    a.slot = b.slot AND
+                    a.key = b.key
+            )
+        ORDER BY
+            account_id
+        ",
+    )?;
+
+    let mut select_fungible_asset_deltas_stmt = conn.prepare_cached(
+        "
+        SELECT
+            account_id, faucet_id, SUM(delta) AS value
+        FROM
+            account_fungible_asset_deltas
+        WHERE
+            account_id IN rarray(?1) AND
+            block_num <= ?2
+        GROUP BY
+            account_id, faucet_id
+        HAVING
+            value != 0
+        ORDER BY
+            account_id
+        ",
+    )?;
+
+    let mut select_latest_non_fungible_assets_stmt = conn.prepare_cached(
+        "
+        SELECT
+            account_id, vault_key
+        FROM
+            account_non_fungible_asset_updates a
+        WHERE
+            account_id IN rarray(?1) AND
+            block_num <= ?2 AND
+            is_remove = 0 AND
+            NOT EXISTS(
+                SELECT 1
+                FROM account_non_fungible_asset_updates b
+                WHERE
+                    a.account_id = b.account_id AND
+                    b.is_remove = 1 AND
+                    b.block_num <= ?2 AND
+                    a.vault_key = b.vault_key AND
+                    a.block_num < b.block_num
+            )
+        ORDER BY
+            account_id
+        ",
+    )?;
+
+    struct AccountRecord {
+        code: AccountCode,
+        nonce: Option<Felt>,
+        storage: BTreeMap<u8, StorageSlot>,
+        assets: Vec<Asset>,
+    }
```

Comment:
Might be worth adding an `impl` to this to separate it from the record iteration?

```rust
impl AccountRecord {
    // We should start by reading the `nonce` then it doesn't have to be optional?
    fn new(id: AccountId, nonce: Nonce) -> Self {..}

    fn push_asset(&mut self, ...)
}
```